src/main/scala/uk/gov/nationalarchives/omega/api/repository/SparqlParams.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2023 The National Archives
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.omega.api.repository
+
+import org.apache.jena.rdf.model.{ Resource, ResourceFactory }
+import uk.gov.nationalarchives.omega.api.messages.request.ListAgentSummary
+import uk.gov.nationalarchives.omega.api.repository.model.AgentTypeMapper
+
+case class SparqlParams(
+  booleans: Map[String, Boolean] = Map.empty,
+  uris: Map[String, String] = Map.empty,
+  dateTimes: Map[String, String] = Map.empty,
+  values: Map[String, List[Resource]] = Map.empty,
+  queryExtension: Option[String] = None
+)
+object SparqlParams extends AgentTypeMapper {
+  def from(listAgentSummary: ListAgentSummary): SparqlParams = {
+    val valuesMap = getValuesMap(listAgentSummary)
+    val uriMap = getUriMap(listAgentSummary)
+    val booleanMap = getBooleanMap(listAgentSummary)
+    val dateTimeMap = getDateTimeMap(listAgentSummary)
```

Comment:
Should we add a test to parameterize `dateTime`

Reply:
I have added a new test suite `SparqlParamsSpec`. Interestingly this led me to realise that `versionTimestamp` is not being correctly interpreted in the SPARQL queries currently as I missed [some of the documentation](https://github.com/nationalarchives/ctd-omega-services/blob/2a683b480bc68795a8315ed2e5c9cbc04a631b3b/schema/omega-services.async.api.yaml#L582C13-L582C13) in the schema which for some reason I can't see in AsyncAPI Studio. I have created [a new ticket](https://national-archives.atlassian.net/browse/PACT-1087) to address this.